test-case/check-alsabat.sh

```diff
@@ -50,39 +56,56 @@ then
 	exit 2
 fi
 
+pcmid_p=${pcm_p:-1}
+
 [[ ${OPT_VALUE_lst['s']} -eq 1 ]] && func_lib_start_log_collect
 
-function __upload_wav_file
-{
-    # upload the alsabat wav file
-    for file in /tmp/bat.wav.*
-    do
-	    size=`ls -l $file | awk '{print $5}'`
-	    if [[ $size -gt 0 ]]; then
-		    cp $file $LOG_ROOT/
-	    fi
-    done
-}
+func_pipeline_export "$tplg" "type:playback & id:$pcmid_p"
+
+# parser the parameters of the specified playback pipeline
+channel=$(func_pipeline_parse_value 0 ch_max)
+rate=$(func_pipeline_parse_value 0 rate)
+fmts=$(func_pipeline_parse_value 0 fmts)
 
 # check the PCMs before alsabat test
 dlogi "check the PCMs before alsabat test"
-[[ $(aplay -Dplug$pcm_p -d 1 /dev/zero -q) ]] && die "Failed to play on PCM: $pcm_p"
-[[ $(arecord -Dplug$pcm_c -d 1 /dev/null -q) ]] && die "Failed to capture on PCM: $pcm_c"
+aplay -Dplug"$pcm_p" -d 1 /dev/zero -q || die "Failed to play on PCM: $pcm_p"
+arecord -Dplug"$pcm_c" -d 1 /dev/null -q || die "Failed to capture on PCM: $pcm_c"
```

Comment:
this looks like an unrelated cleanup?

Reply:
I requested this bug fix, see earlier comments. Agreed it should ideally be in a different commit.

Reply:
i have separated it as a new commit.